packages/app/obojobo-document-engine/src/scripts/viewer/stores/recommendation-store.js

```diff
@@ -0,0 +1,81 @@
+import Common from 'Common'
+import RecommendationUtil from '../../viewer/util/recommendation-util'
+
+const { Dispatcher } = Common.flux
+const { Store } = Common.flux
+
+class RecommendationStore extends Store {
+	constructor() {
+		super('recommendationStore')
+
+		Dispatcher.on(
+			{
+				'question:scoreSet': payload => {
+					RecommendationUtil.onQuestionAnswered(this.state, payload)
+					this.updateRecommendations()
+					this.triggerChange()
+				},
+
+				'question:view': () => {
+					RecommendationUtil.onOpenQuestion(this.state)
+					this.updateRecommendations()
+					this.triggerChange()
+				},
+
+				'nav:gotoPath': payload => {
+					if (!this.state) return
+
+					this.state.currentPageID = payload.value.path
+					RecommendationUtil.onOpenPage(this.state)
+					this.updateRecommendations()
+					this.triggerChange()
+				}
+			},
+			this
+		)
+	}
+
+	updateRecommendations() {
+		this.state.recommendations = RecommendationUtil.createRecommendations(this.state)
+		this.state.overallConfidence = RecommendationUtil.updateOverallConfidence(this.state)
+		return this.state.recommendations
+	}
+
+	init(navState) {
+		console.log(navState)
+		this.state = {
+			confidenceLevels: [],
+			overallConfidence: 0,
+			currentPageID: null,
+			pages: [],
+			recommendations: []
+		}
+
+		let first = true
+		for (const page in navState.itemsByPath) {
+			if (!(navState.itemsByPath[page].contentType === 'Page')) continue
+
+			if (first) {
+				this.state.currentPageID = page
+				first = false
+			}
+
+			this.state.confidenceLevels[page] = 36
```

Comment:
Comments needed to explain parts of the algorithm, such as the relevance behind using "36" here